src/vm/types/mod.rs

```diff
@@ -330,31 +554,126 @@ impl Value {
         //     this is a problem _if_ the static analyzer cannot already prevent
         //     this case. This applies to all the constructor size checks.
         let type_sig = TypeSignature::construct_parent_list_type(&list_data)?;
-        Ok(Value::List(ListData { data: list_data, type_signature: type_sig }))
+        Ok(Value::Sequence(SequenceData::List(ListData { data: list_data, type_signature: type_sig })))
     }
 
     pub fn buff_from(buff_data: Vec<u8>) -> Result<Value> {
         // check the buffer size
         BufferLength::try_from(buff_data.len())?;
         // construct the buffer
-        Ok(Value::Buffer(BuffData { data: buff_data }))
+        Ok(Value::Sequence(SequenceData::Buffer(BuffData { data: buff_data })))
     }
 
     pub fn buff_from_byte(byte: u8) -> Value {
-        Value::Buffer(BuffData { data: vec![byte] })
+        Value::Sequence(SequenceData::Buffer(BuffData { data: vec![byte] }))
+    }
+
+    pub fn string_ascii_from_bytes(bytes: Vec<u8>) -> Result<Value> {
```

Comment:
Can this just error out if `bytes.len()` doesn't convert to a valid `BufferLength`?  No need to parse it if it's gonna fail anyway ;)

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1779/commits/6677278149912738bfe8f7e74c758a7fc4c24d97